delivery/client.py

```diff
@@ -416,6 +416,21 @@ def query_metadata_raw(
 
         return res.json()
 
+    def query_metadata(
+        self,
+        components: collections.abc.Iterable[cm.Component]=(),
+        type: dso.model.Datatype | tuple[dso.model.Datatype]=None,
+    ) -> tuple[dso.model.ArtefactMetadata]:
```

Comment:
it would be nice to add a short docstr, esp. considering parameters (what does it mean if multiple / no components are passed..)